### PR TITLE
Pull down frontend configs

### PR DESCRIPTION
**Pull request by dkniffin**
## Why?

- #94
- It'd be nice to start linting our JS and CSS on rails-only projects as well

## What Changed?

- Add code to pull down the latest of those two configs.

### CODE REVIEW

File: lib/boxcar/app_builder.rb [hunk 1]
```diff
@@ -160,6 +160,14 @@ def create_rubocop_config
       copy_file ".rubocop.yml", ".rubocop.yml"
     end
 
+    def create_eslint_config
+      run "wget https://raw.githubusercontent.com/smashingboxes/web-boilerplate/master/.eslintrc"
+    end
+
+    def create_stylelint_config
+      run "wget https://github.com/smashingboxes/web-boilerplate/blob/master/.stylelint.config.js"
```
Comment (prawl) [inline lib/boxcar/app_builder.rb]: I believe this needs to be a raw version of the file like on line 164.  Without that you're downloading a HTML file.

**Review by prawl — APPROVED** [review]
Might wanna double check stylelint config otherwise shipit

File: lib/boxcar/app_builder.rb (hunk 1 above)
Comment (dkniffin) [reply lib/boxcar/app_builder.rb]: good catch. 👍

**Comment by dkniffin** [comment]
Oh. I guess another good thing to include would be a test that makes sure it can run the linters, and that they pass. And includes those in `.travis.yml`

File: lib/boxcar/app_builder.rb [hunk 2]
```diff
@@ -160,16 +160,33 @@ def create_rubocop_config
...
+    def create_eslint_config
+      run "wget https://raw.githubusercontent.com/smashingboxes/web-boilerplate/master/.eslintrc"
+    end
+
+    def create_stylelint_config
+      run "wget https://raw.githubusercontent.com/smashingboxes/web-boilerplate/master/stylelint.config.js"
+    end
+
+    def setup_package_json
+      remove_file "package.json"
```
Comment (prawl) [inline lib/boxcar/app_builder.rb:172]: What's creating the existing package.json?

File: lib/boxcar/app_builder.rb [hunk 3]
```diff
@@ -160,16 +160,33 @@ def create_rubocop_config
...
-    def cleanup_other_linter_violations
+    def cleanup_other_rubocop_violations
       split_long_comments "config/initializers/backtrace_silencers.rb"
       split_long_comments "config/environments/production.rb"
       split_long_comments "db/seeds.rb"
     end
 
+    def cleanup_eslint_violations
+      eslint_disable_file 'app/assets/javascripts/cable.js'
+    end
```
Comment (prawl) [inline lib/boxcar/app_builder.rb]: Nice, 👏

File: templates/travis.yml.erb [hunk 4]
```diff
@@ -1,6 +1,10 @@
 language: ruby
 rvm:
   - 2.3.0
+before_install:
+  - curl -o- -L https://yarnpkg.com/install.sh | bash -s -- --version version-number
+  - export PATH="$HOME/.yarn/bin:$PATH"
+cache: yarn
```
Comment (prawl) [inline templates/travis.yml.erb:7]: I don't think I've seen caching in our existing projects.  How much does it improve build times?

File: lib/boxcar/app_builder.rb (hunk 2 above)
Comment (dkniffin) [reply lib/boxcar/app_builder.rb:172]: Rails creates an almost empty one. /shrug

**Review by prawl — APPROVED** [review]
LGTM

File: templates/travis.yml.erb (hunk 4 above)
Comment (dkniffin) [reply templates/travis.yml.erb:7]: I think it's built into most of the builds, based on the language. In this case, the language is ruby, so it doesn't know to cache `yarn`

**Pull request merged**